autopilot/choice_test.go

```diff
@@ -0,0 +1,154 @@
+package autopilot
+
+import (
+	"encoding/binary"
+	"math"
+	"testing"
+)
+
+var (
+	nID1 = NodeID([33]byte{1})
+	nID2 = NodeID([33]byte{2})
+	nID3 = NodeID([33]byte{3})
+	nID4 = NodeID([33]byte{4})
+)
+
+// TestChooseNEmptyMap checks that chooseN returns an empty result when no
+// nodes are chosen among.
+func TestChooseNEmptyMap(t *testing.T) {
+	t.Parallel()
+
+	nodes := map[NodeID]*AttachmentDirective{}
+	for i := 0; i < 10; i++ {
```

Comment:
Loop can go away here with usage of `testing/quick`.

Reply:
Done!